university-of-southern-queensland-harvard-agps.csl

```diff
@@ -0,0 +1,399 @@
+      </else>
+    </choose>
+  </macro>
+  <macro name="title">
+    <choose>
+      <if type="chapter paper-conference article-journal article-magazine article-newspaper broadcast interview manuscript map personal_communication speech thesis map post post-weblog" match="none">
+        <text variable="title" font-style="italic" suffix=" "/>
+      </if>
+      <else>
+        <text variable="title" quotes="true" suffix=", "/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="publisher-place">
+    <choose>
+      <if variable="container-title">
+        <choose>
+          <if type="article article-journal article-magazine article-newspaper" match="none">
+            <text variable="publisher-place" suffix=", "/>
+          </if>
+        </choose>
+      </if>
+      <else-if variable="event" match="none">
+        <text variable="publisher-place"/>
+      </else-if>
+    </choose>
+  </macro>
+  <macro name="year-date">
+    <choose>
+      <if variable="issued">
+        <choose>
+          <if type="legal_case" match="any">
+            <date form="text" variable="issued" prefix="(" suffix=")">
+              <date-part name="year"/>
+            </date>
+          </if>
+          <else>
+            <date variable="issued">
+              <date-part name="year"/>
+            </date>
+          </else>
+        </choose>
+      </if>
+      <else>
+        <text term="no date" form="short"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="day-month-date">
+    <date variable="issued">
+      <date-part name="day" form="numeric" suffix=" "/>
+      <date-part name="month" form="long"/>
+    </date>
+  </macro>
+  <macro name="edition">
+    <choose>
+      <if is-numeric="edition">
+        <group delimiter=" " suffix=", ">
+          <number variable="edition" form="ordinal"/>
+          <text term="edition" form="short" strip-periods="true"/>
+        </group>
+      </if>
+      <else>
+        <text variable="edition"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="format">
+    <choose>
+      <if type="song motion_picture broadcast" match="any">
+        <text variable="medium"/>
+      </if>
+      <else>
+        <text variable="genre"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="container">
+    <choose>
+      <if type="chapter paper-conference" match="any">
+        <choose>
+          <if variable="container-title">
+            <choose>
+              <if variable="editor author" match="all"/>
+              <else>
+                <text term="in" suffix=" "/>
+              </else>
+            </choose>
+            <text variable="container-title" font-style="italic" suffix=", "/>
+          </if>
+          <else-if variable="event">
+            <text value="paper presented at "/>
+            <text variable="event"/>
+            <text variable="event-place" prefix=", "/>
+          </else-if>
+        </choose>
+      </if>
+      <else-if type="webpage" match="any">
```

Comment:
typically we also add `post` and `post-weblog` as they usually require the same output as websites. Please add if you see fit.